qiskit/qpy/__init__.py

```diff
@@ -57,16 +62,36 @@
         new_qc = qpy.load(fd)[0]
 
 The :func:`qiskit.qpy.dump` function also lets you
-include multiple circuits in a single QPY file::
+include multiple circuits in a single QPY file:
+
+.. plot::
+    :include-source:
+    :nofigs:
+    :context:
 
     with open('twenty_bells.qpy', 'wb') as fd:
         qpy.dump([qc] * 20, fd)
 
 and then loading that file will return a list with all the circuits
 
+.. plot::
+    :include-source:
+    :nofigs:
+    :context:
+
     with open('twenty_bells.qpy', 'rb') as fd:
         twenty_new_bells = qpy.load(fd)
 
+.. plot::
+    :nofigs:
+    :context:
+
+    # This block is hidden from readers. It's cleanup code.
+    from pathlib import Path
+    Path("bell.qpy").unlink()
+    Path("twenty_bells.qpy").unlink()
```

Comment:
I'm not super wild about the docs build actually writing out dirty files into the working directory while we're going.

Could we either leave the above QPY stuff as `.. code-block`, or alternatively use a prior hidden `context` block to override `open` to be something that secretly returns `io.BytesIO()` objects (potentially wrapped to avoid their context managers clearing them up prematurely).  Obviously that's a complete Python crime if it were real code, but if the context objects are localised and don't leak beyond this file, it'll keep the docs build entirely in-memory, not touching the file-system with temporaries.

Reply:
Yeah that's a good point. The bytes object idea is a good one; I think it should work. Otherwise, I think it's fine not to test difficult cases like these.